packages/xrpl/src/sugar/autofill.ts

```diff
@@ -280,22 +279,21 @@ export async function calculateFeePerTransactionType(
     tx.TransactionType === 'AccountDelete' ||
```

Comment:
The rounding operation a few lines above (on line 275), `baseFee = product.dp(0, BigNumber.ROUND_CEIL)` is not necessary.

We should perform rounding operation after completing all the calculations, otherwise we risk losing precision of values.

Since we are invoking `totalFee.dp(0)` eventually, this rounding operation can be removed.

Here is the docs: https://mikemcl.github.io/bignumber.js/#dp